projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -32,179 +39,243 @@
-
-  // region Lawsuit form fields
-  // General
-  private String spamType = "הודעה אלקטרונית";
-  private String claimAmount = "סכום תביעה";
+  // Lawsuit optional fields
   private String moreThanFiveLawsuits = "הגיש";
   private String lessThanFiveLawsuits = "לא הגיש/ו";
-  private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+  private String claimCaseHaser =
+      "לאחר שהתובע חזר בו מהסכמתו לקבלת דבר/י פרסומת\nמהנתבע/ים, על-ידי משלוח הודעת סירוב כהגדרתה בחוק";
   private String claimCaseSubscription =
-      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
-
-  // User
-  private String userPrivateName = "שם פרטי";
-  private String userLastName = "שם משפחה";
-  private String userID = "ת ז משתמש";
-  private String userAddress = "כתובת משתמש";
-  private String userPhone = "טלפון משתמש";
-  private String userFax = "פקס משתמש";
-
-  // First spam company's details
-  private String companyName = "שם החברה";
-  private String companyId = "מספר ח.פ";
-  private String companyAddress = "כתובת החברה";
-  private String companyPhone = "טלפון החברה";
-  private String companyFax = "פקס חברה";
-
-  // Second spam company's details
-  private String company2Name = "שם החברה 2";
-  private String company2Id = "מספר ח.פ 2";
-  private String company2Address = "כתובת החברה 2";
-  private String company2Phone = "טלפון החברה 2";
-  private String company2Fax = "פקס חברה 2";
-  // endregion
+      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש  \nלקבלת דבר/י הפרסומת";
 
+  EditText receivedDate;
+  DatePickerDialog datePickerDialog;
+  Calendar calendar;
   Button createPdfButton;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_lawsuit_pdf);
-    createPdfButton = (Button) findViewById(R.id.button_createPDF);
 
-    initDirStructure();
+    createPdfButton = findViewById(R.id.button_createPDF);
+    receivedDate = findViewById(R.id.editText_spamDate);
+
     setTimeZones();
+    createPdfButton.setOnClickListener(
+        v -> {
+          checkPermissions();
+          if (permissionGranted) {
+            sharePdf(createPdf());
+          }
+        });
+    receivedDate.setOnClickListener(v -> displayDatePicker());
+  }
 
-    createPdfButton.setOnClickListener(v -> checkPermissionsThenCreatePdf());
+  // Used for date fields in the lawsuit
+  private void setTimeZones() {
+    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
   }
 
-  private void createPdf() {
-    // TODO: Success reading license file only from "res/raw". Perhaps should be read from different
-    // dir
-    // Activate iText license
-    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+  // Save PDF to device. Return absFilename of the file
+  private String createPdf() {
+
+    // Read and fill template
```

Comment:
The function name is "fillTemplate", you don't need to add comment that tell me that it fill the template.
This is why good function and variables names are great, you don't need to comment what it do.

Reply:
This comment is not for describing what `fillTemplate` does, but rather as another step mentioning in the overall "guidance" of how and what `createPdf` goes through.

Same and as part of _"Draw lines to PDF"_ ,  _"Save file to external storage"_ comments.

Added these as to what we spoke about in our last conversation, about wrapping the code with more general "guidance" comments, to ease understanding of what's going on in the function overall.

But yea, now that you mention, it actually doesn't add any additional information...
So thanks :)

Done.